File: ESTRUTURAS/1260/main.dart
import 'dart:io';
import 'dart:collection';

SplayTreeMap<String, int> getTrees() {
  SplayTreeMap<String, int> trees = SplayTreeMap<String, int>();
  String? tree;
  while (!(tree = stdin.readLineSync()?.trim() ?? '').isEmpty) {
    trees[tree] = trees.putIfAbsent(tree, () => 0) + 1;
  }
  return trees;
}

void formattedPrint(int c, SplayTreeMap<String, int> trees) {
  if (c > 0) print('');
  int total = trees.values.fold(0, (a, b) => a + b);
  for (String tree in trees.keys) {
    print('$tree ${(100 * trees[tree]! / total).toStringAsFixed(4)}');
  }
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  stdin.readLineSync();
  for (int c = 0; c < n; c++) {
    formattedPrint(c, getTrees());
  }
}

File: ESTRUTURAS/1260/main.js
function getTrees() {
    let line, trees = {};
    while (line = lines[i++]) {
        let tree = line.trim();
        if (!(tree in trees)) trees[tree] = 0;
        trees[tree]++;
    }
    return trees;
}

function formattedPrint(c, trees) {
    if (c) console.log();
    let total = Object.values(trees).reduce((a, b) => a + b);
    for (let tree of Object.keys(trees).sort())
        console.log(`${tree} ${(100 * trees[tree] / total).toFixed(4)}`);
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');
let i = 0; n = parseInt(lines[i++]);
i++;
for (let c = 0; c < n; c++) {
    formattedPrint(c, getTrees());
}
